src/TypeScript/EslintBridgeClient/EslintBridgeStartUp.cs

```diff
@@ -19,17 +19,183 @@
+        private readonly INodeLocator nodeLocator;
+        private readonly IFileSystem fileSystem;
+        private readonly ILogger logger;
+        private TaskCompletionSource<int> startTask;
+
+        internal Process Process;
+
+        [ImportingConstructor]
+        public EslintBridgeStartUp([Import(EslintBridgeDirectoryMefContractName)] string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            ILogger logger)
+            : this(eslintBridgeStartupScriptPath, nodeLocator, new FileSystem(), logger)
+        {
+        }
+
+        internal EslintBridgeStartUp(string eslintBridgeStartupScriptPath,
+            INodeLocator nodeLocator,
+            IFileSystem fileSystem,
+            ILogger logger)
+        {
+            this.eslintBridgeStartupScriptPath = eslintBridgeStartupScriptPath;
+            this.nodeLocator = nodeLocator;
+            this.fileSystem = fileSystem;
+            this.logger = logger;
+        }
+
+        public Task<int> Start()
+        {
+            if (startTask != null && !startTask.Task.IsFaulted)
+            {
+                return startTask.Task;
+            }
+
+            startTask = new TaskCompletionSource<int>();
+            try
+            {
+                StartServer();
+            }
+            catch (Exception ex)
+            {
+                startTask.SetException(ex);
+            }
+
+            return startTask.Task;
+        }
+
+        public void Dispose()
+        {
+            Stop();
+        }
+
+        private void StartServer()
+        {
+            var nodePath = FindNodePath();
+            var serverStartupScriptLocation = FindServerScriptFile();
+
+            var psi = new ProcessStartInfo
+            {
+                FileName = nodePath,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                UseShellExecute = false, // required if we want to capture the error output
+                ErrorDialog = false,
+                CreateNoWindow = true,
+                Arguments = serverStartupScriptLocation
+            };
+
+            psi.EnvironmentVariables.Add("NODE_PATH", @"%APPDATA%\npm\node_modules");
+
+            Process = new Process {StartInfo = psi};
+            Process.ErrorDataReceived += OnErrorDataReceived;
+            Process.OutputDataReceived += OnOutputDataReceived;
+
+            Process.Start();
+            logger.WriteLine(Resources.INFO_ServerProcessId, Process.Id);
+            logger.LogDebug(Resources.INFO_ServerHasExisted, Process.HasExited);
+
+            Process.BeginErrorReadLine();
+            Process.BeginOutputReadLine();
+        }
+
+        private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(e.Data))
+            {
+                return;
+            }
+
+            logger.LogDebug("ESLINT-BRIDGE OUTPUT: " + e.Data);
```

Comment:
I left the log-debug strings as plain strings, didn't think it's worth putting them in resources. WDYT?

Reply:
Ok; the other messages should be resources, though, with a standard message prefix.